src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -179,24 +193,30 @@ private static void RestPostMediaType(String restPostMediaType) throws Subscript
     }
 
     /**
-     * Validation of email address Throws SubscriptionValidationException if validation of the parameter
-     * fails due to wrong format of parameter.
+     * Validation of email address Throws SubscriptionValidationException if
+     * validation of the parameter fails due to wrong format of parameter.
      *
      * @param email
      */
     public static void validateEmail(String email) throws SubscriptionValidationException {
-        // When this regExp need to be changed then remember to change the one in the
-        // back-end (validEmailRegExpression in subscription.js), which do the same
-        // email validation check. The two
-        // regEx always need to be the same for ensuring the same check.
-        String validEmailRegExpression = "^(([^<>()\\[\\]\\\\.,;:\\s@\"]+(\\.[^<>()\\[\\]\\\\.,;:\\s@\"]+)*)|(\".+\"))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
-        final Pattern VALID_EMAIL_ADDRESS_REGEX = Pattern.compile(validEmailRegExpression,
-                Pattern.CASE_INSENSITIVE);
-        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(email);
-        if (!(matcher.matches())) {
+        String validEmailRegEx = null;
+        try {
+            validEmailRegEx = RegExProvider.getRegExs().getString("validEmail");
+        } catch (JSONException | IOException e) {
+            LOGGER.error("Error message: " + e.getMessage(), e);
+        }
+
+        if (validEmailRegEx == null || validEmailRegEx.isEmpty()) {
             throw new SubscriptionValidationException(
-                    "Notification type is set to [MAIL] but the given notificatioMeta contains an invalid e-mail ["
-                            + email + "]");
+                    "A valid regular expression for subscription email validation is not provided");
+        } else {
```

Comment:
This else is not needed, since if above if-statement is true then it will raise an exception and not run the code below.

Reply:
done